consensus/polybft/wallet/key.go

```diff
@@ -20,40 +20,44 @@ func NewKey(raw *Account) *Key {
 	}
 }
 
+// String returns hex encoded ECDSA address
 func (k *Key) String() string {
 	return k.raw.Ecdsa.Address().String()
 }
 
+// Address returns ECDSA address
 func (k *Key) Address() ethgo.Address {
 	return k.raw.Ecdsa.Address()
 }
 
-func (k *Key) Sign(b []byte) ([]byte, error) {
-	s, err := k.raw.Bls.Sign(b)
+// Sign signs the provided digest with BLS key
+func (k *Key) Sign(digest []byte) ([]byte, error) {
+	signature, err := k.raw.Bls.Sign(digest)
 	if err != nil {
 		return nil, err
 	}
 
-	return s.Marshal()
+	return signature.Marshal()
 }
 
-// SignEcdsaMessage signs the proto message with ecdsa
-func (k *Key) SignEcdsaMessage(msg *proto.Message) (*proto.Message, error) {
-	raw, err := protobuf.Marshal(msg)
+// SignIBFTMessage signs the IBFT consensus message with ECDSA key
+func (k *Key) SignIBFTMessage(msg *proto.Message) (*proto.Message, error) {
+	msgRaw, err := protobuf.Marshal(msg)
```

Comment:
So the `Message` contains the `Signature` too? I know that technically it should not be a problem since `Signature` should be empty. However, this is traditionally done with two structures. A `Container` and a `SignedContainer`:
```
type Container struct { // Message
   Field1 uint64
   Field2 uint64
}

type SignedContainer struct {
    Container []byte
    Signature [32]byte
}
```
It is also easier to know this way the lifecycle of the objects (i.e. you get a `SignedContainer` you have to decode it, from that point onwards you can use the `Container`).

Reply:
Ok, agree. However it is not strictly related to this PR, since I believe go-ibft needs to be changed?

Reply:
Yup, also, it would mean breaking backward compatibility.